plugin/src/Settings.lua

```diff
@@ -0,0 +1,78 @@
+--[[
+	Persistent plugin settings.
+]]
+
+local plugin = plugin or script:FindFirstAncestorWhichIsA("Plugin")
+local Rojo = script:FindFirstAncestor("Rojo")
+
+local Log = require(Rojo.Log)
+
+local defaultSettings = {
+	openScriptsExternally = false,
+	twoWaySync = false,
+	showNotifications = true,
+	playSounds = true,
+}
+
+local Settings = {}
+
+Settings._values = table.clone(defaultSettings)
+Settings._updateListeners = {}
+
+if plugin then
+	for name, defaultValue in pairs(Settings._values) do
+		local savedValue = plugin:GetSetting("Rojo_" .. name)
+
+		if savedValue == nil then
+			-- plugin:SetSetting hits disc instead of memory, so it can be slow. Spawn so we don't hang.
+			task.spawn(plugin.SetSetting, plugin, "Rojo_" .. name, defaultValue)
```

Comment:
I don't think `plugin.SetSetting` yields. If it does blocking I/O, then spawning will help Rojo initialize a little quicker, but Studio will still pause while the settings are actually loaded.